src/utils/__tests__/files.test.ts

```diff
@@ -28,6 +28,9 @@ describe('withTempDir', () => {
       }, cleanupEnabled);
     } finally {
       if (cleanupEnabled) {
+        // We intentionally do not block on the clean up operation
+        // so wait ~100ms before checking
+        await new Promise(resolve => setTimeout(resolve, 100));
```

Comment:
Sleep is never a good idea. I mean, sleep is a great idea ... but not here. No better way? Is this related to the await removals that @tonyo is asking about?

Reply:
>Sleep is never a good idea. I mean, sleep is a great idea ... but not here.

Totally agree.

>No better way?

We can try polling with a... timeout?

>Is this related to the await removals that @tonyo is asking about?

Yup :(